subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan_aux.c

```diff
@@ -683,25 +685,29 @@ static void isr_rx(struct lll_scan *lll, struct lll_scan_aux *lll_aux,
 	}
 
 isr_rx_do_close:
-	if (lll_aux) {
-		radio_isr_set(isr_done, NULL);
-	} else {
-		/* Send message to flush Auxiliary PDU list */
-		if (err != -ECANCELED) {
-			struct node_rx_pdu *node_rx;
+	/* In case of any error we need to send message to flush aux context,
+	 * otherwise we already sent a PDU and can decide in ULL what to do.
+	 */
+	if (err != -ECANCELED) {
+		struct node_rx_pdu *node_rx;
 
-			node_rx = ull_pdu_rx_alloc();
-			LL_ASSERT(node_rx);
+		node_rx = ull_pdu_rx_alloc();
+		LL_ASSERT(node_rx);
 
-			node_rx->hdr.type = NODE_RX_TYPE_EXT_AUX_RELEASE;
+		node_rx->hdr.type = NODE_RX_TYPE_EXT_AUX_RELEASE;
 
-			node_rx->hdr.rx_ftr.param = lll->lll_aux;
+		node_rx->hdr.rx_ftr.param = lll_aux ? (void *)lll_aux :
+						      (void *)lll;
 
-			ull_rx_put(node_rx->hdr.link, node_rx);
-			ull_rx_sched();
-		}
+		ull_rx_put(node_rx->hdr.link, node_rx);
+		ull_rx_sched();
+	}
```

Comment:
I approve this. Yes, ULL (ticker started aux reception) and LLL (scan context rx ISR) scheduled reception failure both should lead to auxiliary context release.